api/commands/validate.ts

```diff
@@ -1,16 +1,51 @@
 import * as cronParser from "cron-parser";
 import { dataform } from "@dataform/protos";
+import * as fs from "fs";
+import * as path from "path";
 
+const SCHEDULES_JSON_PATH = "schedules.json";
 const EMAIL_REGEX = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
 
+interface IOptions {
+  projectDir?: string;
+}
+
 function validateEmail(email: string) {
   return EMAIL_REGEX.test(email);
 }
 
-export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, compiledGraph: dataform.ICompiledGraph) {
+export function validateSchedulesFile(
+  compiledGraph: dataform.ICompiledGraph,
+  options: IOptions
+): string[] {
+  if (!options || !options.projectDir) {
+    return [];
+  }
+  const filePath = path.resolve(path.join(options.projectDir, SCHEDULES_JSON_PATH));
+  if (!fs.existsSync(filePath)) {
```

Comment:
this is OK for me, but I think we should rename the function to `validateSchedulesFileIfExists` - as a caller i'd expect this function to fail if it couldn't read the relevant file.